src/main/scala/io/iohk/ethereum/db/storage/BlockHeadersStorage.scala

```diff
@@ -24,7 +25,23 @@ class BlockHeadersStorage(val dataSource: DataSource) extends KeyValueStorage[Bl
   override def valueDeserializer: (IndexedSeq[Byte]) => BlockHeader =
     (encodedBlockHeader: IndexedSeq[Byte]) => rlpDecode[BlockHeader](encodedBlockHeader.toArray)
 
-  override protected def apply(dataSource: DataSource): BlockHeadersStorage = new BlockHeadersStorage(dataSource)
+  override protected def apply(dataSource: DataSource): BlockHeadersStorage =
+    new BlockHeadersStorage(dataSource, blockHeadersNumbersStorage)
+
+  def get(blockNumber: BigInt): Option[BlockHeader] = if (blockNumber == 0) {
+    //todo get block from config, or we will put 0 block to storage at app startup if it is not there?
+    None
+  } else {
+    blockHeadersNumbersStorage.get(blockNumber).flatMap(key => super.get(key))
+  }
+
+  override def update(toRemove: Seq[BlockHeaderHash], toUpsert: Seq[(BlockHeaderHash, BlockHeader)]): BlockHeadersStorage = {
```

Comment:
No, but in Json-RPC yes.

what about the following taken from [PV61](https://github.com/ethereum/wiki/wiki/Ethereum-Wire-Protocol#pv61-specific) 

> BlockHashesFromNumber [+0x08: P, number: P, maxBlocks: P] Requires peer to reply with a BlockHashes message. Message should contain block with that of number number on the canonical chain. Should also be followed by subsequent blocks, on the same chain, detailing a number of the first block hash and a total of hashes to be sent. Returned hash list must be ordered by block number in ascending order.